store.cpp

```diff
@@ -6,10 +6,38 @@
 
 #include <glog/logging.h>
 
+Store::Store(std::optional<std::string>& filename) {
+  if (filename != std::nullopt) {
+    std::ifstream fin;
+    std::string line;
+    std::string key;
+    std::string val;
+    fin.open(filename.value());
+    while (fin) {
+      getline(fin, line);
+      int pos = line.find(':');
+      key = line.substr(0, pos);
+      val = line.substr(pos + 1);
+      umap_[key] = val;
```

Comment:
It seems like you're only adding the last value to the key on reading from file. What if multiple values are assigned to the same key? Correct me if I have misunderstood something.

Reply:
all values associated with a key will be stored in the file in the format, "key":"value1 value2 ...". All values will be loaded.

Reply:
Okay, I see. My misunderstanding was assuming that your umap_ was an unordered map of string to vector but it is a map of string to strings.